src/PimApi/PimCatalogApiClient.php

```diff
@@ -219,4 +227,30 @@ public function getMappedProducts(
 
         return $response['_embedded']['items'];
     }
+
+    /**
+     * @return array<mixed>
+     *
+     * @throws CatalogDisabledException
+     * @throws PimApiException
+     * @throws PimApiUnauthorizedException
+     */
+    public function getMappedProduct(string $catalogId, string $productUuid): array
+    {
+        $pimUrl = $this->getPimUrl();
+
+        $catalogEndpointUrl = "$pimUrl/api/rest/v1/catalogs/$catalogId/mapped-products/$productUuid";
+
+        $response = $this->getClient()->request('GET', $catalogEndpointUrl);
+
+        $this->throwOnErrorCode(200, $response->getStatusCode(), "Couldn't get mapped product");
+
+        $response = $response->toArray();
+
+        if (isset($response['message']) || isset($response['error'])) {
+            throw new CatalogDisabledException();
+        }
+
+        return $response;
+    }
```

Comment:
I think that you can factorize the getCatalogProduct method. In my opinion, there is a quiet big duplicate code section here. What do you think @akeneo/squad-octopus ?

Reply:
We could factorize the majority of this page but I think we want the code as straightforward as possible. What is your pov @RiffVentura @masacc  ?

Reply:
What kind of refactoring do you have in mind?
Keep in mind that this whole class is temporary and will be scrapped as soon as [api-php-client](https://github.com/akeneo/api-php-client) supports all of our new catalog-related endpoints

Reply:
I disagree to factorize this for all the reasons Etienne and Rifat evoked + in a CQRS way of thinking, it's not such a bad thing to have duplicate code if the contexts are different which is the case right here.a

Reply:
Agree with the 3 others, DRY & code mutualization is only valid when **you are sure** that both pieces of code will always evolve the same way.

Here, they both answer to different needs and there is a high probably that `getMappedProduct` will evolve in a different direction, if there more errors to handle for example.

Reply:
For the fun fact, mutualization of code is a trap we have fallen in for the PIM in the past.
This is why some legacy parts of the PIM are hard to work with.

We have proved in our more recent features that avoiding DRY principles, even if we have to slightly copy/paste some code is better in the long term.
I really like the current way of working of the octopus where we postpone mutualization for later to have a better view of how it should be organized.